tests/test_aggregates.py

```diff
@@ -0,0 +1,264 @@
+
+    def test_with_decorator_without_pre_initialization(self):
+        @Aggregate()
+        def test_function(x):
+            return x
+
+        assert getattr(test_function, '_flow_aggregate', False)
+
+    def test_groups_of_invalid_num(self):
+        invalid_values = [{}, 'str', -1, -1.5]
+        for invalid_value in invalid_values:
+            with pytest.raises((TypeError, ValueError)):
+                Aggregate.groupsof(invalid_value)
+
+    def test_group_by_invalid_key(self):
+        with pytest.raises(TypeError):
+            Aggregate.groupby(1)
+
+    def test_groupby_with_valid_type_default_for_Iterable(self, setUp, project):
+        Aggregate.groupby(['half', 'even'], default=[-1, -1])
+
+    def test_groupby_with_invalid_type_default_key_for_Iterable(self, setUp, project):
+        with pytest.raises(TypeError):
+            Aggregate.groupby(['half', 'even'], default=-1)
+
+    def test_groupby_with_invalid_length_default_key_for_Iterable(self, setUp, project):
+        with pytest.raises(ValueError):
+            Aggregate.groupby(['half', 'even'], default=[-1, -1, -1])
+
+
+# Test the _StoreAggregate and _StoreDefaultAggregate classes
+class TestStoreAggregate(AggregateProjectSetup):
+
+    def test_valid_aggregator_non_partial(self, setUp, project):
+        # Return groups of 1
+        def helper_aggregator(jobs):
+            for job in jobs:
+                yield (job, )
+
+        aggregate_instance = Aggregate(helper_aggregator)
+
+        aggregate_instance = aggregate_instance._create_StoreAggregates()
+        aggregate_job_manual = helper_aggregator(project)
+        aggregate_instance(project)  # Generate Aggregates
+
+        assert [aggregate for aggregate in aggregate_job_manual] == \
+               list(aggregate_instance)
+
+    def test_valid_aggregator_partial(self, setUp, project):
+        aggregate_instance = Aggregate(lambda jobs: [jobs])
+        aggregate_instance = aggregate_instance._create_StoreAggregates()
+        aggregate_instance(project)  # Generate Aggregates
+
+        assert [tuple([job for job in project])] == \
+               list(aggregate_instance)
+
+    def test_valid_sort_by(self, setUp, project):
+        helper_sort = partial(sorted, key=lambda job: job.sp.i)
+        aggregate_instance = Aggregate(sort_by='i')
+        aggregate_instance = aggregate_instance._create_StoreAggregates()
+        aggregate_instance(project)  # Generate Aggregates
+        assert [tuple(helper_sort(project))] == list(aggregate_instance)
+
+    def test_valid_reversed_sort(self, setUp, project):
+        helper_sort = partial(sorted, key=lambda job: job.sp.i, reverse=True)
+        aggregate_instance = Aggregate(sort_by='i', reverse_order=True)
+        aggregate_instance = aggregate_instance._create_StoreAggregates()
+        aggregate_instance(project)  # Generate Aggregates
+        assert [tuple(helper_sort(project))] == list(aggregate_instance)
```

Comment:
It may be worth creating a helper function that does the comparison between a function's and an `_StoreAggregate`'s aggregates given a `FlowProject`.